pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1203,7 +1207,13 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundTCPListenerOptsForPort
 func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(node *model.Proxy, listenerOpts buildListenerOpts,
 	pluginParams *plugin.InputParams, listenerMap map[string]*outboundListenerEntry,
 	virtualServices []model.Config, actualWildcard string) {
-
+	if features.BlockHTTPonHTTPSPort {
+		if listenerOpts.port == CanonicalHttpsPort && pluginParams.Port.Protocol == protocol.HTTP {
+			msg := fmt.Sprintf("listener conflict detected: service %v specifies an HTTP service on HTTPS only port %d.", pluginParams.Service.Hostname, CanonicalHttpsPort)
```

Comment:
didnt you have a PR to not do sprintf?

Reply:
the problem is needless sprintfs that happen millions of times per second. I'm not sure we need to blanket ban sprintf. Especially since ever log message is using this